int64 mask_size
int64 n_mask
int64 range
float64 threshold
float64 mu1
float64 mu2
int64 sample_step
int64 ntotal_sample
